OpenSim/Simulation/Model/TwoFrameLinker.h

```diff
@@ -252,6 +266,33 @@ TwoFrameLinker<C, F>::TwoFrameLinker(const std::string &name,
 
 template <class C, class F>
 TwoFrameLinker<C, F>::TwoFrameLinker(const std::string &name,
+    const F& frame1, const SimTK::Transform& transformInFrame1,
+    const F& frame2, const SimTK::Transform& transformInFrame2)
+    : TwoFrameLinker()
+{
+    this->setName(name);
+
+    PhysicalOffsetFrame frame1Offset(frame1.getName() + "_offset",
+        frame1, transformInFrame1);
+
+    PhysicalOffsetFrame frame2Offset(frame2.getName() + "_offset",
+        frame2, transformInFrame2);
+
+    // Append the offset frames to the Joints internal list of frames
+    int ix1 = append_frames(frame1Offset);
+    int ix2 = append_frames(frame2Offset);
+    this->finalizeFromProperties();
```

Comment:
The doxygen for `Component::finalizeFromProperties()` should explain that it is expected to call `Component::finalizeFromProperties()` in scenarios like this. What specifically does this do that enables you to call the lines below?